static/app/utils/useKeyPress.tsx

```diff
@@ -1,34 +1,31 @@
-// hook from https://usehooks.com/useKeyPress/
 import {useEffect, useState} from 'react';
 
+/**
+ * Hook to detect when a specific key is being pressed
+ */
 function useKeyPress(targetKey: React.KeyboardEvent['key']) {
-  // State for keeping track of whether key is pressed
   const [keyPressed, setKeyPressed] = useState(false);
 
-  // If pressed key is the target key then set to true
-  function downHandler({key}) {
-    if (key === targetKey) {
-      setKeyPressed(true);
+  useEffect(() => {
+    function downHandler({key}: KeyboardEvent) {
+      if (key === targetKey) {
+        setKeyPressed(true);
+      }
     }
-  }
 
-  // If released key is the target key then set to false
-  function upHandler({key}) {
-    if (key === targetKey) {
-      setKeyPressed(false);
+    function upHandler({key}: KeyboardEvent) {
+      if (key === targetKey) {
+        setKeyPressed(false);
+      }
     }
-  }
-
-  // Add event listeners
-  useEffect(() => {
     window.addEventListener('keydown', downHandler);
     window.addEventListener('keyup', upHandler);
-    // Remove event listeners on cleanup
+
     return () => {
       window.removeEventListener('keydown', downHandler);
       window.removeEventListener('keyup', upHandler);
     };
-  }, []); // Empty array ensures that effect is only run on mount and unmount
+  }, [targetKey]);
```

Comment:
now always when `targetKey` changes, we will be adding the same `EventListener` over and over... do we want this?

Reply:
Only whenever `targetKey` changes. We've only been using `useKeyPress` with `targetKey` hardcoded to a string.

Reply:
Yeah, it will clean up the old event listeners too. This is a good change

Reply:
in this case it's fine :)